Modules/_testcapimodule.c

```diff
@@ -865,36 +868,44 @@ test_L_code(PyObject *self, PyObject *Py_UNUSED(ignored))
         return NULL;
 
     num = PyLong_FromLong(42);
-    if (num == NULL)
-        return NULL;
+    if (num == NULL) {
+        goto error;
+    }
 
     PyTuple_SET_ITEM(tuple, 0, num);
 
     value = -1;
     if (!PyArg_ParseTuple(tuple, "L:test_L_code", &value)) {
-        return NULL;
+        goto error;
     }
-    if (value != 42)
-        return raiseTestError("test_L_code",
+    if (value != 42) {
+        raiseTestError("test_L_code",
             "L code returned wrong value for long 42");
+        goto error;
+    }
 
-    Py_DECREF(num);
     num = PyLong_FromLong(42);
```

Comment:
The following code is redundant. In Python 2 there were tests for `PyInt_FromLong` and `PyLong_FromLong`.